### PR TITLE
Add a dimmer model with subdevices 

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
Blocked behind #92

**Comment by peternewman** [comment]
Fix lint errors. Resync to fix Travis.

File: firmware/src/dimmer_model.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * @addtogroup rdm_models
+ * @{
+ * @file dimmer_model.h
+ * @brief A dimmer-only RDM Model.
+ *
+ * This model simulates a dimmer with different subdevices, one per dimmer
+ * module. Each sub-device consumes one slot of DMX data. The model implements
+ * many of the PIDs from E1.37-1.
```
Comment (peternewman) [inline firmware/src/dimmer_model.h]: s/many/some/ unless you're adding lots more code.

File: firmware/src/rdm.h [hunk 2]
```diff
@@ -98,6 +98,26 @@ static const uint16_t MAX_DMX_START_ADDRESS = 512;
...
+
+/**
+ * @brief Indicates the responder has subdevices.
+ */
+static const uint16_t MUTE_SUBDEVICE_FLAG = 0x02;
+
+/**
+ * @brief Indicates the responder is in bootloader mode.
+ */
+static const uint16_t MUTE_BOOTLOADER_FLAG = 0x02;
```
Comment (peternewman) [inline firmware/src/rdm.h]: 4

File: firmware/src/rdm.h [hunk 3]
```diff
@@ -98,6 +98,26 @@ static const uint16_t MAX_DMX_START_ADDRESS = 512;
...
+
+/**
+ * @brief Indicates the responder is a proxy.
+ */
+static const uint16_t MUTE_PROXY_FLAG = 0x02;
```
Comment (peternewman) [inline firmware/src/rdm.h]: 8

File: firmware/src/rdm_responder.c [hunk 4]
```diff
@@ -365,18 +370,19 @@ int RDMResponder_SetUnMute(const RDMHeader *header) {
...
     return RDM_RESPONDER_NO_RESPONSE;
   }
-  g_responder.is_muted = false;
+  g_responder->is_muted = false;
 
   ReturnUnlessUnicast(header);
 
   uint8_t *ptr = g_rdm_buffer + sizeof(RDMHeader);
-  ptr = PushUInt16(ptr, 0);  // set control fields to 0
+  ptr = PushUInt16(ptr,
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: These two functions still feel unnecessarily duplicated.

File: firmware/src/rdm_responder.c [hunk 5]
```diff
@@ -392,6 +398,9 @@ int RDMResponder_GetSupportedParameters(const RDMHeader *header,
       case PID_SOFTWARE_VERSION_LABEL:
       case PID_DMX_START_ADDRESS:
       case PID_IDENTIFY_DEVICE:
+        if (g_responder->is_subdevice) {
```
Comment (peternewman) [inline firmware/src/rdm_responder.c]: How come you're returning this?

**Comment by peternewman** [comment]
Generally LGTM, just a few comments.

File: firmware/src/dimmer_model.h (hunk 1 above)
Comment (nomis52) [inline firmware/src/dimmer_model.h]: It should all be here by the end of the weekend

File: firmware/src/rdm.h (hunk 2 above)
Comment (nomis52) [inline firmware/src/rdm.h]: Good catch!

File: firmware/src/rdm.h (hunk 3 above)
Comment (nomis52) [inline firmware/src/rdm.h]: Fixed

File: firmware/src/rdm_responder.c (hunk 4 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: Yup, more cleanup coming later once the additional tests are in.

File: firmware/src/rdm_responder.c (hunk 5 above)
Comment (nomis52) [inline firmware/src/rdm_responder.c]: Section 2 of E1.37-1 "Sub-devices should always declare all their supported parameters in their list of Supported Parameters. "

I've opened a bug to add a RDM Responder tests for this issue.

Comment (peternewman) [inline firmware/src/rdm_responder.c]: I think we need to open a bug in E1.20 too then, section 10.4.1 says "PIDs that are included in the minimum support list, indicated by the “Required” column of Table A-3, shall not be reported.". Incidentally "This document does not change or modify the requirements stated in ANSI E1.20. Requirements stated in this document are in addition to the stated ANSI E1.20 requirements." that from E1.37-1 seems a bit contradictory, there are some requirements from E1.20 which adding to would involve changing, e.g. do/don't report required parameters.

Comment (nomis52) [inline firmware/src/rdm_responder.c]: It's already on the fix-it list for E1.20 :)

**Comment by peternewman** [comment]
LGTM

**Pull request merged**